src/test/java/com/exasol/sql/expression/function/exasol/ExasolAggregateFunctionTest.java

```diff
@@ -0,0 +1,43 @@
+package com.exasol.sql.expression.function.exasol;
+
+import static com.exasol.hamcrest.SqlFragmentRenderResultMatcher.rendersTo;
+import static com.exasol.sql.expression.ExpressionTerm.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+import com.exasol.sql.StatementFactory;
+import com.exasol.sql.dql.select.Select;
+import com.exasol.sql.expression.BooleanTerm;
+
+public class ExasolAggregateFunctionTest {
+    @Test
+    void testAggregateFunctionCoalesce() {
+        final Select select = StatementFactory.getInstance().select() //
+                .function(ExasolAggregateFunction.APPROXIMATE_COUNT_DISTINCT, "COUNT_APPR", column("customer_id"));
+        select.from().table("orders");
+        select.where(BooleanTerm.gt(column("price"), integerLiteral(1000)));
+        assertThat(select,
+                rendersTo("SELECT APPROXIMATE_COUNT_DISTINCT(customer_id) COUNT_APPR FROM orders WHERE price > 1000"));
+    }
+
+    @Test
+    void testAggregateFunctionCount() {
+        final Select select = StatementFactory.getInstance().select() //
+                .function(ExasolAggregateFunction.COUNT, keyWord("DISTINCT"), column("customer_id"));
+        select.from().table("orders");
+        assertThat(select, rendersTo("SELECT COUNT( DISTINCT customer_id) FROM orders"));
```

Comment:
Yes, it's always there. I have two spaces before and after the keywords. The logic to control it seems quite complicated so I decided to leave them be like that for now because it doesn't affect a query too much